solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -592,18 +620,33 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     }
   }
 
-  private void tagRequestWithRequestId(ResponseBuilder rb) {
-    final boolean ridTaggingDisabled = rb.req.getParams().getBool(CommonParams.DISABLE_REQUEST_ID, false);
-    if (! ridTaggingDisabled) {
-      String rid = getOrGenerateRequestId(rb.req);
-      if (StringUtils.isBlank(rb.req.getParams().get(CommonParams.REQUEST_ID))) {
-        ModifiableSolrParams params = new ModifiableSolrParams(rb.req.getParams());
-        params.add(CommonParams.REQUEST_ID, rid);//add rid to the request so that shards see it
-        rb.req.setParams(params);
+  private static final List<String> VERBOSE_LOGGING_PARAMS = Arrays.asList("fq", "json");
+  private SolrParams removeVerboseParams(final SolrParams params) {
+    // Filter params by removing kv of VERBOSE_LOGGING_PARAMS, so that we can then call toString
+    SolrParams filteredParams = new SolrParams() {
+      @Override
+      public Iterator<String> getParameterNamesIterator() {
+        return Iterators.filter(params.getParameterNamesIterator(), s -> !VERBOSE_LOGGING_PARAMS.contains(s));
       }
-      if (rb.isDistrib) {
-        rb.rsp.addToLog(CommonParams.REQUEST_ID, rid); //to see it in the logs of the landing core
+
+      @Override
+      public String get(String param) {
```

Comment:
This method and below do not have the filtering effect?

Reply:
Good catch! 

I actually copied this from https://github.com/fullstorydev/lucene-solr/blob/release/8.8/solr/core/src/java/org/apache/solr/core/SolrCore.java#L2712 , which the coder appears to be aware of the possible issue.

Since this filtered params will only be used in the logging logic, which calls `toString` that calls `getParameterNamesIterator` which in theory should not call `get` on keys that do not exist, but I agree this kind of assumptions could break as it depends on how the caller make use of the filtered `SolrParams`.

It's a bit of performance vs code completeness - what do you think? :)

Reply:
Yeah, if this is an object which has to be used in some particular way, i.e., iterate parameter names, and call get, then it is probably fine. I think it does not hurt to add a `contains` check on `param`, either.

Reply:
Let's keep this simple for now since this will be called on every single KVs.

Thanks for bringing that up! 😊